src/Microsoft.AspNetCore.Mvc.RazorPages/NonHandlerAttribute.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages
+{
+    /// <summary>
+    /// Specifies that the targeted method is not a page handler method.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+    public class NonHandlerAttribute : Attribute
```

Comment:
I haven't identified a place to put one of these yet, but we always end up needing things like this.